fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -186,6 +188,15 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
         GlobalScope.launch(Dispatchers.IO) {
             setStartupMetrics(store, settings())
         }
+
+        val applicationContext = this.applicationContext
+        MainScope().launch {
+            ProcessLifecycleOwner.get().lifecycle.addObserver(
+                ClientDeduplicationLifecycleObserver(
```

Comment:
The lifecycle observer that I added is almost a direct copy of what Glean uses under the hood and I include a link to show that. I did this so that it is more evident that we are copying the scheduling of the baseline ping. If it isn't a hard requirement I might just leave that as is for now? I may go back in and update the comments in that observer to make the tie to the baseline ping scheduling more evident.

Reply:
Sure, that's fine. Likely, once the investigation is complete, we can remove this ping, or refactor the observer then.